client/model.go

```diff
@@ -40,6 +40,73 @@ type ProjectCreatePayload struct {
 	Description string `json:"description"`
 }
 
+type EnvironmentCreate struct {
+	Name                        string         `json:"name"`
+	ProjectId                   string         `json:"projectId"`
+	DeployRequest               *DeployRequest `json:"deployRequest"`
+	WorkspaceName               string         `json:"workspaceName,omitempty"`
+	RequiresApproval            bool           `json:"requiresApproval,omitempty"`
+	ContinuousDeployment        bool           `json:"continuousDeployment,omitempty"`
+	PullRequestPlanDeployments  bool           `json:"pullRequestPlanDeployments,omitempty"`
+	AutoDeployOnPathChangesOnly bool           `json:"autoDeployOnPathChangesOnly,omitempty"`
+	AutoDeployByCustomGlob      bool           `json:"autoDeployByCustomGlob,omitempty"`
+}
+
+type DeployRequest struct {
+	BlueprintId          string                `json:"blueprintId,omitempty"`
+	BlueprintRevision    string                `json:"blueprintRevision,omitempty"`
+	BlueprintRepository  string                `json:"blueprintRepository,omitempty"`
+	ConfigurationChanges *ConfigurationChanges `json:"configurationChanges,omitempty"`
+	TTL                  *TTL                  `json:"ttl,omitempty"`
+	EnvName              string                `json:"envName,omitempty"`
+	UserRequiresApproval bool                  `json:"userRequiresApproval,omitempty"`
+	Targets              string                `json:"targets,omitempty"`
+	PrPlanOnly           bool                  `json:"prPlanOnly,omitempty"`
+	// TODO: not sure about this one
```

Comment:
not sure about the type, gonna test it when I add deployments on update

Reply:
maybe `map[string]interface{}` ?
https://stackoverflow.com/questions/40559250/golang-dynamically-creating-member-of-struct

Reply:
I will change it on the pr for [this ticket](https://trello.com/c/K68z3UXy/4784-terraform-provider-environment-resource-part-2) that way I'll also actually test it